app/service/data_svc.py

```diff
@@ -131,3 +136,24 @@ def __init__(self, dao):
               'ON a.ability_id=b.id ' \
               'WHERE a.op_id = %s;' % op_id
         return await self.dao.raw_select(sql)
+
+def make_uuid(_filename):
+    uuid_string = _filename.split('/')[-1].split('.')[0]
+    try:
+        val = UUID(uuid_string, version=4)
+        return _filename
+    except ValueError:
+        _uuid = str(uuid4())
+        new_ability_file = _filename.split('/')[:-1]
+        new_ability_file.append('{}.yml'.format(_uuid))
+        new_ability_file = '/'.join(new_ability_file)
```

Comment:
this looks like it copies the file, right? can we have it replace the file instead?